### PR TITLE
bug fix #1716

### LINKED ISSUE #1716: bug: Removing/Changing position of Lights [L] in editor doesn't function correctly
### Description

I'm not sure which update this bug took place (my bad for not reporting sooner) but sometime in 7.1

Editing a map with Lights already placed, will cause the incorrect light to be deleted/moved.
This bug also happens when moving lights - both on single/multi layer edits.

It actually looks like it's also duplicating the light tile as well:
https://user-images.githubusercontent.com/110916959/215350302-55c9b03c-fe5c-47b5-8c30-7b979366a9aa.mp4

### Steps to Reproduce

1. Add light to map via editor
2. save map
3. Try to update those lights

### Version with bug

v0.7.1.88

### Last version that worked well

Bug was introduced sometime in 7.1 I believe.  A few months back at least.

### Affected platforms

Windows, macOS

### Did you find any workaround?

no

### Relevant log output

_No response_

### Duplicate Bug Check

- [X] This bug report is not a duplicate to the best of my knowledge.

### PR DESCRIPTION
**Pull request by TimtyG**
Fixes #1716

I've added TileX and TileY to both methods GetHashCode() & Equals() in LightBase.cs

This change ensure that the hash code and equality comparison take into account the TileX and TileY properties, necessary for drag / copy / undo ...

### CODE REVIEW

File: Intersect (Core)/GameObjects/LightBase.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ Color color
         /// <returns></returns>
         public override int GetHashCode()
         {
-            return Intensity ^ (int)Expand ^ (Color?.GetHashCode() ?? 0);
+            return Intensity ^ (int)Expand ^ (Color?.GetHashCode() ?? 0) ^ TileX.GetHashCode() ^ TileY.GetHashCode();
```
Comment (lodicolo) [inline Intersect (Core)/GameObjects/LightBase.cs:71]: Change to `HashCode.Combine(...)`, don't pass `GetHashCode()` into the `Combine(...)` method

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect (Core)/GameObjects/LightBase.cs (hunk 1 above)
Comment (TimtyG) [reply Intersect (Core)/GameObjects/LightBase.cs:71]: return HashCode.Combine(Intensity, (int)Expand, Color, TileX, TileY);

Will work for you Panda?

Comment (lodicolo) [reply Intersect (Core)/GameObjects/LightBase.cs]: @TimtyG Don't cast to int here, just pass Expand directly into the method

Comment (TimtyG) [reply Intersect (Core)/GameObjects/LightBase.cs]: Alright, on it

**Review by lodicolo — APPROVED** [review]
Approved, waiting on autobuild before merging

**Pull request merged**